src/base/array_storage.rs

```diff
@@ -299,3 +300,14 @@ where
         self.as_slice().iter().fold(0, |acc, e| acc + e.extent())
     }
 }
+
+// TODO: Where to put this impl block?
+impl<T, const R: usize, const C: usize> SMatrix<T, R, C> {
+    /// Creates a new statically-allocated matrix from the given [ArrayStorage].
+    #[inline(always)]
+    pub const fn from_array_storage(storage: ArrayStorage<T, R, C>) -> Self {
+        // This is sound because the row and column types are exactly the same as that of the
+        // storage, so there can be no mismatch
+        unsafe { Self::from_data_statically_unchecked(storage) }
+    }
+}
```

Comment:
I would suggest to put this on `matrix.rs`, bellow the impl block containing `from_data_statically_unchecked`.
Perhaps we should add a `TODO` comment saying that this should be removed in favor of `Matrix::from_data` once `const-fn` support generic trait bounds.